scala-package/core/src/main/scala/org/apache/mxnet/IO.scala

```diff
@@ -160,6 +161,107 @@ class DataBatch(val data: IndexedSeq[NDArray],
   def provideLabel: ListMap[String, Shape] = providedLabel
 }
 
+object DataBatch {
+  /**
+   * Builder class for DataBatch.
+   */
+  class Builder() {
+    private var data: IndexedSeq[NDArray] = null
```

Comment:
nulls are not recommended in Java/Scala ? can you not use empty? `IndexedSeq.empty`? same for others?

Reply:
no it behaves the same as `DataBatch`

Reply:
null check for `DataBatch` are here and there in Scala frontend, e.g., https://github.com/apache/incubator-mxnet/blob/master/scala-package/core/src/main/scala/org/apache/mxnet/module/DataParallelExecutorGroup.scala#L490
We need to keep Builder behave the same. `DataBatch.label == null` and `DataBatch.label == IndexedSeq.empty` are different in `Exec.forward`

Reply:
ok thanks for explaining.